src/meshes/p4est_mesh.jl

```diff
@@ -1595,19 +1597,19 @@ end
 
 # Copy global quad ID to quad's user data storage, will be called below
 function save_original_id_iter_volume(info, user_data)
-  info_obj = unsafe_load(info)
+  info_pw = PointerWrapper(info)
 
   # Load tree from global trees array, one-based indexing
-  tree = unsafe_load_tree(info_obj.p4est, info_obj.treeid + 1)
+  tree = unsafe_load_tree(info_pw.p4est, info_pw.treeid[] + 1)
   # Quadrant numbering offset of this quadrant
   offset = tree.quadrants_offset
   # Global quad ID
-  quad_id = offset + info_obj.quadid
+  quad_id = offset + info_pw.quadid[]
 
   # Unpack quadrant's user data ([global quad ID, controller_value])
-  ptr = Ptr{Int}(unsafe_load(info_obj.quad.p.user_data))
+  pw = PointerWrapper(Int, pointer(info_pw.quad.p.user_data))
```

Comment:
For constructs like this it might be useful to have a function like
```julia
PointerWrapper(::Type{T}, pw::PointerWrapper) where T = PointerWrapper{T}(pointer(pw))
```
What do you think?

Reply:
What is `info_pw.quad.p.user_data` - a `PointerWrapper` wrapping a `Ptr{Cvoid}`? If so, that sounds reasonable to me